homeassistant/components/device_tracker/huawei_hilink.py

```diff
@@ -0,0 +1,217 @@
+"""
+Support for Huawei HiLink routers.
```

Comment:
HiLink is the name of their app. From app description:

Huawei HiLink APP can be used on the following devices:
Mobile WiFi (E5 series): 
E5331, E5332, E5372, E5375, E5756
E5151, E5220, E5221, E5251, E589
E5730, E5776, E5377, E5786, E5573
EC5321, EC5377U, E5771s
HWD34,HWD35

Wingles
E8231, E8278, EC315, E355

CPEs
E5186, E5170, B310, B315s,HWS31

Home routers:
WS318, WSR20, WS331a, WS331b, WS330, WS880, WS326, WS328, Honor Cube(WS860), WS831

Reply:
So this code should be merged with the other platform. The platform should figure out if it can use the new API or the old one.